File: src/main/java/edu/cnm/deepdive/sieve/bitset/Sieve.java
package edu.cnm.deepdive.sieve.bitset;

import java.util.BitSet;

public class Sieve {

  private static BitSet sieve(int limit) {
    // TODO Implement with a BitSet for keeping track of the candidates.
    throw new UnsupportedOperationException();
  }

  public static void main(String... args) {
    int upperBound = 1_000_000;
    long start = System.currentTimeMillis();
    BitSet primes = sieve(upperBound);
    long end = System.currentTimeMillis();
    System.out.printf("Java Sieve with BitSet:%n"
            + "%1$,d primes found between %2$,d and %3$,d in %4$d ms.%n",
        primes.cardinality(), primes.nextSetBit(0), primes.previousSetBit(primes.size() - 1),
        end - start);
  }

}

File: src/main/kotlin/edu/cnm/deepdive/sieve/bitset/sieve.kt
package edu.cnm.deepdive.sieve.bitset

import java.util.BitSet

fun sieve(limit: Int): BitSet {
    // TODO Implement with a BitSet for keeping track of the candidates.
    throw UnsupportedOperationException()
}

fun main(args: Array<String>) {
    val start = System.currentTimeMillis()
    val upperBound = 1_000_000
    val primes = sieve(upperBound)
    val end = System.currentTimeMillis()
    println(
        """
            Kotlin Sieve with BitSet: 
            ${primes.cardinality()} primes found between ${primes.nextSetBit(0)} and ${primes.previousSetBit(primes.size() - 1)} (inclusive) in ${end - start} ms.
        """.trimIndent()
    )
}